src/sqlfluff/core/rules/base.py

```diff
@@ -955,6 +956,26 @@ class RuleManifest:
     rule_class: Type[BaseRule]
 
 
+@dataclass
+class RulePack:
+    """A bundle of rules to be applied.
+
+    This contains a set of rules, post filtering but also contains the mapping
+    required to interpret any noqa messages found in files.
+
+    The reason for this object is that rules are filtered and instantiated
+    into this pack in the main process when running in multi-processing mode so
+    that user defined rules can be used without reference issues.
+    """
+
+    rules: List[BaseRule]
+    reference_map: Dict[str, List[str]]
```

Comment:
Can you add a comment and/or example of the contents of this dictionary? It would make the code a lot more understandable. As written, I'm unsure:
* Whether the rule code is a key or a value
* Why multiple values are supported

Is ordering important for the values (i.e. the use of `List` rather than `Set`)?

Reply:
Ah good point. I think a `Set` would be more appropriate. I'll add some docs to explain exactly what it is.